tests/unit/test_charm.py

```diff
@@ -138,6 +144,23 @@ def test_default_scrape_config_is_always_set(self):
         prometheus_scrape_config = scrape_config(config, "prometheus")
         self.assertIsNotNone(prometheus_scrape_config, "No default config found")
 
+    def test_honor_labels_is_always_set_in_scrape_configs(self):
+        rel_id = self.harness.add_relation(RELATION_NAME, "provider")
+        self.harness.add_relation_unit(rel_id, "provider/0")
+        self.harness.update_relation_data(
+            rel_id,
+            "provider",
+            {
+                "scrape_metadata": json.dumps(SCRAPE_METADATA),
+                "scrape_jobs": json.dumps(DEFAULT_JOBS),
+            },
+        )
+
+        prometheus_scrape_config = yaml.safe_load(self.harness.charm._prometheus_config())
+        for job in prometheus_scrape_config["scrape_configs"]:
+            self.assertIn("honor_labels", job)
```

Comment:
This line could probably be removed - KerError would fail the test anyway.

Reply:
I see no harm in it being a two step process -- Assert key exists then Assert key has expected value.